osu.Game/Rulesets/Mods/Mod.cs

```diff
@@ -69,7 +69,7 @@ public abstract class Mod : IMod, IJsonSerializable
         /// <summary>
         /// Creates a copy of this <see cref="Mod"/> initialised to a default state.
         /// </summary>
-        public virtual Mod CreateCopy() => (Mod)Activator.CreateInstance(GetType());
+        public virtual Mod CreateCopy() => (Mod)MemberwiseClone();
```

Comment:
This should likely be in a different PR, kind-of out of scope for what this is implementing.

Reply:
I think it's fine here, without this any mods with internal state would be thoroughly broken.

Reply:
This code is about fixing an issue when doing something to the mods as discussed, this should've been in another PR, just saying for the author to take note of that later on.